src/Diagrams/TwoD/Path/Boolean.hs

```diff
@@ -241,6 +244,32 @@ loopExclusion :: Double -> FillRule
               -> [Located (Trail' Loop V2 Double)]
               -> [Located (Trail' Loop V2 Double)]
               -> [Located (Trail' Loop V2 Double)]
-loopExclusion tol fill path1 path2 =
+loopExclusion tol fill path1 path2 = normalizeWindings fill $
   map path2loop $ C.exclusion (map loop2path path1)
   (map loop2path path2) (fillrule fill) tol
+
+-- Force all top level loops to wind counterclockwise and revese inner loops as needed.
+normalizeWindings :: FillRule -> [Located (Trail' Loop V2 Double)] -> [Located (Trail' Loop V2 Double)]
+normalizeWindings fill = concat . map forceCC . nestedGroups fill where
+  forceCC ls | (l:_) <- ls, isClockwise l = map reverseLocLoop ls
+             | otherwise                  = ls
+
+-- Group a list of loops such that the first element of each group contains all the others.
+nestedGroups :: FillRule -> [Located (Trail' Loop V2 Double)] -> [[Located (Trail' Loop V2 Double)]]
+nestedGroups fill = map flatten . go [] where
+  go ts               []                      = ts
+  go []               (l:ls)                  = go [Node l []]                      ls
+  go ((Node n ns):ts) (l:ls) | l `contains` n = go [Node l [Node n ns]]             ls
+                             | n `contains` l = go (Node n (go ns [l]) : ts)        ls
+                             | otherwise      = go (Node n ns          : go ts [l]) ls
+  contains = containsBy fill
+
+-- To test if s contains t, we can merely test if any point of t is within s
+-- because the binary operations guarantee that the loops do not intersect.
+containsBy :: FillRule -> Located (Trail' Loop V2 Double) -> Located (Trail' Loop V2 Double) -> Bool
+containsBy Winding s t = isInsideWinding s (atStart t)
+containsBy EvenOdd s t = isInsideEvenOdd s (atStart t)
```

Comment:
This seems destined to corner cases where the start is right on the edge.  This seems like a rather common case if these trails came from intersecting.  I think we need some comments laying out the context of what loops we could be getting at this point.

Reply:
Not entirely sure what you mean by "right on the edge". Assuming you mean that (atStart t) is a point on s, in that case the binary operation will have produced a single loop, not two separate loops.

Reply:
Again, I'm not entirely clear on the context, but I'm thinking of something like the [`exclusion` example in the manual](https://archives.haskell.org/projects.haskell.org/diagrams/doc/manual.html#boolean-operations-on-paths).  The corners of the interior loop are on the edge of the outer loop.

Reply:
You're right. It's definitely possible to detect this case, but resolving it does not seem straight forwards. I'l have to think about it.

Reply:
I know how to fix this. Now the fun part is that it relies on an unmerged changeset I've been working on for IntersectionExtras and it creates a circular module dependency with that changeset. Time to go finish that and break it out into two modules.

Reply:
Good luck!